### ISSUE TITLE
script-servo is broken

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang-nursery/rustc-perf#289]
```
...
error[E0597]: borrowed value does not live long enough
   --> collector/benchmarks/script-servo/target/debug/build/script-e1750e7c06f94680/out/Bindings/XMLDocumentBinding.rs:827:10
    |
827 |         &InterfaceConstructorBehavior::throw(),
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ temporary value does not live long enough
...
830 |         3);
    |          - temporary value only lives until here
    |
    = note: borrowed value must be valid for the static lifetime...

error[E0597]: borrowed value does not live long enough
    --> collector/benchmarks/script-servo/target/debug/build/script-e1750e7c06f94680/out/Bindings/XMLHttpRequestBinding.rs:1950:10
     |
1950 |         &InterfaceConstructorBehavior::call(_constructor),
     |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ temporary value does not live long enough
...
1953 |         2);
     |          - temporary value only lives until here
     |
     = note: borrowed value must be valid for the static lifetime...

error[E0597]: borrowed value does not live long enough
    --> collector/benchmarks/script-servo/target/debug/build/script-e1750e7c06f94680/out/Bindings/XMLHttpRequestEventTargetBinding.rs:1024:10
     |
1024 |         &InterfaceConstructorBehavior::throw(),
     |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ temporary value does not live long enough
...
1027 |         1);
     |          - temporary value only lives until here
     |
     = note: borrowed value must be valid for the static lifetime...

error[E0597]: borrowed value does not live long enough
   --> collector/benchmarks/script-servo/target/debug/build/script-e1750e7c06f94680/out/Bindings/XMLHttpRequestUploadBinding.rs:601:10
    |
601 |         &InterfaceConstructorBehavior::throw(),
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ temporary value does not live long enough
...
604 |         2);
    |          - temporary value only lives until here
    |
    = note: borrowed value must be valid for the static lifetime...

error: aborting due to 259 previous errors
```

There are lots of these and I haven't really attempted to narrow down the exact problem so I can't say much more; if we don't think this is readily fixable then I can start exploring alternatives. I can reproduce with master of this repository with `cargo check --manifest-path collector/benchmarks/script-servo/Cargo.toml --no-default-features`

cc @nikomatsakis @pnkfelix  -- possibly related to borrowck changes? Though I don't quite see how that could be as the crate doesn't seem to be using NLL.

**Comment by pnkfelix** [comment]
Can you bisect to a nightly ?

**Comment by Mark-Simulacrum** [comment]
Regression appears to be between 2018-10-04 and 2018-10-05 (rust-lang/rust@5597ee8a6483238421750fce4eb7031b12852428 and rust-lang/rust@8c4ad4e9e42f4bcbdbb784692b936c2462061abc; https://github.com/rust-lang/rust/compare/5597ee8a6483238421750fce4eb7031b12852428...8c4ad4e9e42f4bcbdbb784692b936c2462061abc)

**Comment by Mark-Simulacrum** [comment]
@pnkfelix Do you think you'll get a chance to look at this or should I try to allocate some time myself?

**Comment by mati865** [comment]
script-servo now fails with:
```
error[E0119]: conflicting implementations of trait `dom::bindings::trace::JSTraceable` for type `fn(&_) -> _`:
   --> components/script/dom/bindings/trace.rs:435:1
    |
428 | unsafe impl JSTraceable for fn(A) -> B {
    | -------------------------------------------- first implementation here
...
435 | unsafe impl<\'a, A, B> JSTraceable for fn(&A) -> B {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `fn(&_) -> _`
    |
    = note: this behavior recently changed as a result of a bug fix; see rust-lang/rust#56105 for details
```

**Comment by nnethercote** [comment]
I just pushed a fix for this.
